imagen_pytorch/data.py

```diff
@@ -23,7 +31,73 @@ def convert_image_to(img_type, image):
         return image.convert(img_type)
     return image
 
-# dataset and dataloader
+# dataset, dataloader, collator
+
+class Collator:
+    def __init__(self, image_size, url_label, text_label, image_label, name):
+        self.url_label = url_label
+        self.text_label = text_label
+        self.image_label = image_label
+        self.download = url_label is not None
+        self.name = name
+        self.transform = T.Compose([
+            T.Resize(image_size),
+            T.CenterCrop(image_size),
+            T.ToTensor(),
+        ])
+    def __call__(self, batch):
+
+        texts = []
+        masks = []
+        images = []
+        for item in batch:
+            try:
+                if self.download:
+                    image = self.fetch_single_image(item[self.url_label])
+                else:
+                    image = item[self.image_label]
+                image = self.transform(image.convert('RGB'))
+            except:
+                continue
+
+            text, mask = t5.t5_encode_text([item[self.text_label]], return_attn_mask=True, name=self.name)
+            texts.append(text)
+            masks.append(mask)
```

Comment:
you don't actually have to manage the `mask` anymore, as the framework automatically derives it from the text embeddings https://github.com/lucidrains/imagen-pytorch/blob/main/imagen_pytorch/imagen_pytorch.py#L2201

Reply:
From my trainings the result are different when I don't use masks